packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts

```diff
@@ -113,28 +123,41 @@ export class QueryDataViewService {
     return this.currentPanel.webview;
   }
 
-  protected onDidRecieveMessageHandler(message: DataViewEvent) {
+  protected onDidRecieveMessageHandler(message: DataViewEvent): void {
     const { type, format } = message;
     switch (type) {
       case 'activate':
         this.updateWebviewWith(this.queryData);
         break;
       case 'save_records':
-        this.handleSaveRecords(format!);
+        this.handleSaveRecords(format as FileFormat);
         break;
       default:
-        console.log('unknown message type from data view');
+        const errorMessage = nls.localize('error_unknown_error', type);
+        channelService.appendLine(errorMessage);
+        trackErrorWithTelemetry('data_view_message_type', type).catch(
+          console.error
+        );
         break;
     }
   }
 
-  protected handleSaveRecords(format: FileFormat) {
-    const fileService = new FileService(
-      this.queryData,
-      format,
-      getDocumentName(this.document)
-    );
-    fileService.save();
+  protected handleSaveRecords(format: FileFormat): void {
+    try {
+      const fileService = new FileService(
+        this.queryData,
+        format,
+        getDocumentName(this.document)
+      );
+      fileService.save();
+    } catch (err) {
+      const message = nls.localize(
+        'error_data_view_save',
+        QUERY_RESULTS_DIR_NAME
+      );
+      vscode.window.showErrorMessage(message);
+      trackErrorWithTelemetry('data_view_save', message);
```

Comment:
There must be some OS issue, permissions, or unknown, that would cause the file save issue.

<img width="469" alt="save-results-failure" src="https://user-images.githubusercontent.com/599418/101827301-5ef97900-3aed-11eb-9a41-77fd3aa3f2a8.png">